### DISCUSSION TITLE
Add code block to the hero page

### DISCUSSION THREAD

**Discussion by peterfox** [vuejs/vitepress#1753] [Q&A]
I want to add a simple code block to the Hero page of my docs at https://feature-flags.docs.ylsideas.co/

I already have a custom theme based on the default theme. I can edit [VPHome.vue](https://github.com/ylsideas/feature-flags-docs/blob/main/.vitepress/theme/ylsideas/components/VPHero.vue#L30) to include code, but I don't know how to get the styles/functionality that is in the markdown docs.

How would I go about adding a code block with the same styles and the copy button etc., to that component?

My best guess is maybe I need to add a component that then converts the contents to Markdown, or there's simply a component I need to add to the page and configure it to show bash highlighting and pass in the contents.

**Comment by minenwerfer** [comment]
I don't know why this discussion hasn't been moved forward. Having a code sample in the hero page to showcase something is pretty standard at this point. I also wanted to use this in my website, I tried to hack around using builtin CSS classes but I found it to be counterproductive.

Ideally the [Markdown function](https://github.com/vuejs/vitepress/blob/main/src/node/markdown/plugins/containers.ts) should be exposed in the public API to allow custom usage. Something along these lines:

```vue
---
layout: home
---

<script setup>
import { createCodeGroup } from 'vitepress'
const codeGroup = createCodeGroup()
</script>

<template>
  <div v-html="code-group"></div>
</template>
```

@brc-dd do you propose any other solution?

**Reply by brc-dd** [reply]
It's already supported 👀 

````md
---
layout: home
---

```sh
composer require ...
```
````

**Reply by minenwerfer** [reply]
This is only suported in single-column layouts, though, no? In my understanding something like the following isn't well supported at the moment, or is it? https://wasp-lang.dev/

I think my example wasn't clear enough.

**Reply by brc-dd** [reply]
Ah yeah that's currently not supported. Only images are supported there currently.

**Comment by johannes-lindgren** [comment]
@peterfox and @minenwerfer, since you can [include Vue code](https://vitepress.dev/guide/using-vue) in the markdown documents, you can use a a [Teleport](https://vuejs.org/guide/built-ins/teleport#teleport) to inject a Vue.js component into the hero. I can share an example that I created. The code is by no means perfect, but you can modify it to suit your needs:

2. A component `WithinHero.vue` that teleports its slot into the hero, and also overrides the hero's styles so that the content fits:  https://github.com/johannes-lindgren/pure-parse/blob/e2f9d89507d23721073a09b29c0533ee1712cadd/packages/pure-parse/docs/components/WithinHero.vue
3. A component `CodeBlock.vue` that renders text content with syntax highlighting (using highlight.js): https://github.com/johannes-lindgren/pure-parse/blob/e2f9d89507d23721073a09b29c0533ee1712cadd/packages/pure-parse/docs/components/CodeBlock.vue

Don't forget to install `highlight.js` to `devDependencies`.

Then use it as such in `index.md`:

```vue
---
layout: home

hero:
  name: PureParse
# ...
---

<script setup>
import CodeBlock from "/components/CodeBlock.vue";
import WithinHero from "/components/WithinHero.vue";

const code = `
type User = {
  id: number
  name: string
}

const parseUser = object<User>({
  id: parseNumber,
  name: parseString,
})
`
</script>

<WithinHero>
    <CodeBlock :code="code">
      <template #caption>Declare the type:</template>
    </CodeBlock>
</WithinHero>
```

Result:

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/38791b92-7dcc-4744-989b-18db9f1a9cc2">
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/ffac91b9-5aae-45dd-a586-94140779f412">

**Reply by codywakeford** [reply]
Good shout
